hazelcast/include/hazelcast/client/query/predicates.h

```diff
@@ -64,6 +64,15 @@ namespace hazelcast {
                     out_stream.write<int32_t>(static_cast<int32_t>(sizeof...(values)));
                     out_stream.write_objects(values...);
                 }
+
+                template<typename T>
+                multi_predicate(const std::string attribute_name, hazelcast_client &client, const std::vector<T> &values) : base_predicate(client) {
```

Comment:
Why is the `hazelcast_client` needed to create a predicate? IMO it would be great to get rid of it if possible.

Reply:
unfortunately, we need it to access the serialization service. I also do not like it but i had to do it due to the fact that i needed to prepare the binary during construction.

Reply:
I see. I wish we could defer binary construction. Taking a client as a parameter has more problems, like would passing a different to `multi_predicate` than the one calling the actual query cause problems? But I guess this is not something that can be changed in this PR.

Reply:
Passing a different client, yes, that may cause a problem if serialization configs are different. Alternative was to use the client as the factory method to create the `in_predicate` but it again has the same problem if the user would use a client predicate with the other client. I am open to any better solution but could not find one as of yet. Yes, we can open another issue to discuss this topic.